src/MongoDB.Driver.Core/Core/Clusters/Cluster.cs

```diff
@@ -222,9 +222,9 @@ private void RapidHeartbeatTimerCallback(object args)
 
         protected abstract void RequestHeartbeat();
 
-        protected void OnDescriptionChanged(ClusterDescription oldDescription, ClusterDescription newDescription)
+        protected void OnDescriptionChanged(ClusterDescription oldDescription, ClusterDescription newDescription, bool shouldClusterDescriptionChangedEventBePublished)
```

Comment:
Do we need a better name for this variable? Perhaps something related to SDAM Monitoring?
`shouldSdamMonitoringClusterDescriptionChangedEventBePublished`? A bit long, but it might help prevent the confusion that happened below?

Reply:
I will think about it

Reply:
@vincentkam , I'm not sure whether `monitoring` is exactly the sdam part that has been skipped via this `shouldClusterDescriptionChangedEventBePublished` property. In other words, can we say that `SDAM` (server discovery and monitoring) includes two parts: 
1. Server discovery => everything in sdam flow except user-level(diagnostic) events?
2. Monitoring => only is about user-level(diagnostic) events?

if not so, I would just add a comment similar to: it was decided to suppress only sdam events and do not touch sdam flow itself.

Reply:
I may be wrong, but my current understanding of the flow is that any event we raise that is backed by an `Action` is related to SDAM Monitoring, and that anything that is backed by an `EventHandler` is related to the actual SDAM flow/logic. In other words, I _think_ that `shouldClusterDescritionChangedEventBePublished` is merely supressing the SDAM Monitoring event.

Reply:
Notably, the [spec ticket](https://github.com/mongodb/specifications/commit/44efef18ea13cc1c201104052984ccd0b16da462) associated with the drivers ticket seems to only contain changes to the monitoring tests.

Reply:
Since the PR is approved, let's continue discussing this case in the scope of new ticket which will be created for https://jira.mongodb.org/browse/DRIVERS-1249 :)